### PR TITLE
Sleeping is now a status effect

### PR DESCRIPTION
**Pull request by ChangelingRain**
1/4 Complete

### CODE REVIEW

File: code/datums/status_effects/debuffs.dm [hunk 1]
```diff
@@ -1,5 +1,64 @@
...
+		owner.update_stat()
+	if(iscarbon(owner)) //to avoid repeated istypes
+		carbon_owner = owner
+	if(ishuman(owner))
+		human_owner = owner
+	return ..()
+
+/datum/status_effect/incapacitating/sleeping/tick()
+	if(owner.staminaloss)
+		owner.adjustStaminaLoss(-0.35) //reduce stamina loss by 0.35 per tick, 7 per 2 seconds
```
Comment (ChangelingRain) [inline code/datums/status_effects/debuffs.dm]: Should this not update health, and update health when the status effect is removed?

**Comment by Haevacht** [comment]
Comment.

File: code/datums/status_effects/debuffs.dm (hunk 1 above)
Comment (lzimann) [inline code/datums/status_effects/debuffs.dm]: I think it's better to adjust like this, makes more sense than as an one time thing imo

**Review by lzimann — APPROVED** [review]
Looks good overall.

**Comment by lordpidey** [comment]
This looks like it might break His Grace, if you put  it on a table and pick it up repeatedly.  Could you test that?

**Comment by Cyberboss** [comment]
@lordpidey Please elaborate how?

File: code/datums/status_effects/status_effect.dm [hunk 2]
```diff
@@ -17,7 +17,7 @@
 		owner = new_owner
 	if(owner)
 		LAZYADD(owner.status_effects, src)
-	addtimer(CALLBACK(src, .proc/start_ticking), 1) //Give us time to set any variables
+	addtimer(CALLBACK(src, .proc/start_ticking), 0) //Give us the minimum possible time to set any variables
```
Comment (ChangelingRain) [inline code/datums/status_effects/status_effect.dm]: I'm pretty sure you can do var changes before this is called, right @Cyberboss?

Comment (Cyberboss) [inline code/datums/status_effects/status_effect.dm]: The timer will 100% not start until the current thread sleeps, so yeah

Comment (ChangelingRain) [inline code/datums/status_effects/status_effect.dm]: Plus looking over the code the his grace status effect doesn't care about which his grace you're holding, only that you're holding one.

**Pull request merged**